### ISSUE TITLE
Add support for pod affinity

### ISSUE THREAD

**Issue by bskim45** [neo4j/helm-charts#32]
Currently, the helm chart only supports pod anti-affinity. It would be nice to support [pod affinity](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity), which is useful to schedule Pods and PV within the specific AZ.

ex. 
```yaml
podSpec:
  podAntiAffinity: true

  podAffinity:  # default {}
    nodeAffinity:
      requiredDuringSchedulingIgnoredDuringExecution:
        nodeSelectorTerms:
        - matchExpressions:
          - key: failure-domain.beta.kubernetes.io/zone
            operator: In
            values:
              - us-central1-a
```

**Comment by harshitsinghvi22** [comment]
Hi @bskim45 

Thanks for raising this issue. This feature will definitely be worked upon and released in the upcoming future releases
Appreciate your time and effort in raising this issue.

Thanks,
Harshit

**Comment by andrea-berling** [comment]
Any updates on this feature request? I would also be interested in it: I'm trying to use multiple zones for the persistent disks in a causal cluster for better fault tolerance, and I can not find a way to do that. Right now my deployment is failing because a cluster core pod is being scheduled onto a different node from the one that it should ideally be scheduled to in order to be in the same zone as the persistent disk the pod is supposed to use. I thought that node affinity would have been a good solution, but it is not currently supported by the chart.

**Comment by harshitsinghvi22** [comment]
@bskim45  @andrea-berling u can use nodeSelector to achieve the same. 
We just released nodeAffinity with helm charts 4.4.10
Hope this helps
https://github.com/neo4j/helm-charts/releases/tag/4.4.10

Let us know if you still need podAffinity feature , if not please close this issue if the recent updates solve your query

**Comment by harshitsinghvi22** [comment]
@bskim45 , Let us know if you still need podAffinity feature , if not please close this issue if the recent updates solve your query
